napari/_qt/dialogs/preferences_dialog.py

```diff
@@ -0,0 +1,229 @@
+import json
+
+from qtpy.QtCore import Signal
+from qtpy.QtWidgets import (
+    QDialog,
+    QHBoxLayout,
+    QLabel,
+    QListWidget,
+    QPushButton,
+    QStackedWidget,
+    QVBoxLayout,
+    QWidget,
+)
+
+from ..._vendor.qt_json_builder.qt_jsonschema_form import WidgetBuilder
+from ...utils.settings import SETTINGS
+from ...utils.settings._defaults import ApplicationSettings, PluginSettings
+from ...utils.translations import translator
+
+trans = translator.load()
+
+
+class PreferencesDialog(QDialog):
+    """Preferences Dialog for Napari user settings."""
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+
+        self._list = QListWidget(self)
+        self._stack = QStackedWidget(self)
+
+        # Set up buttons
+        self._button_cancel = QPushButton(trans._("Cancel"))
+        self._button_ok = QPushButton(trans._("OK"))
+        self._default_restore = QPushButton(trans._("Restore defaults"))
+
+        # Setup
+        self.setWindowTitle(trans._("Preferences"))
+
+        # Layout
+        main_layout = QHBoxLayout()
+        main_layout.addWidget(self._list)
+        main_layout.addWidget(self._stack)
+
+        buttons_layout = QHBoxLayout()
+        buttons_layout.addWidget(self._button_cancel)
+        buttons_layout.addWidget(self._button_ok)
+
+        layout = QVBoxLayout()
+        layout.addLayout(main_layout)
+        layout.addWidget(self._default_restore)
+        layout.addLayout(buttons_layout)
+
+        self.setLayout(layout)
+
+        # Signals
+
+        self._list.currentRowChanged.connect(
+            lambda index: self._stack.setCurrentIndex(index)
+        )
+        self._button_cancel.clicked.connect(self.on_click_cancel)
+        self._button_ok.clicked.connect(self.on_click_ok)
+        self._default_restore.clicked.connect(self.restore_defaults)
+
+        # Make widget
+
+        self.make_dialog()
+        self._list.setCurrentRow(0)
+
+    def make_dialog(self):
+        """Removes settings not to be exposed to user and creates dialog pages."""
+
+        settings_list = [ApplicationSettings(), PluginSettings()]
+        cnt = 0
+        for key, setting in SETTINGS.schemas().items():
+
+            schema = json.loads(setting['json_schema'])
+            # need to remove certain properties that will not be displayed on the GUI
+            properties = schema.pop('properties')
+            values = setting['model'].dict()
+            for val in settings_list[cnt].NapariConfig().preferences_exclude:
+                properties.pop(val)
+                values.pop(val)
+
+            cnt += 1
+            schema['properties'] = properties
+
+            self.add_page(schema, values)
+
+    def restore_defaults(self):
+        """Launches dialog to confirm restore settings choice."""
+
+        widget = ConfirmDialog(
```

Comment:
using `dialog.show()` here allows for the possibility that this dialog can stay open well beyond it's usefulness (it can get hidden, etc...).  Unless you already considered it and rejected it, this seems like a good time to use `dialog.exec_()`, which will force the user to answer the question... and you can perhaps just use one of the [static members of `QMessageBox`](https://doc.qt.io/qt-5/qmessagebox.html#static-public-members).  Is that something you already considered and rejected?

Reply:
changing to `.exec_()` makes sense.  Just pushed this change.